### ISSUE TITLE
Unable to include a reference to a video

### ISSUE THREAD

**Issue by allenmanning** [quarto-dev/quarto-cli#5847]
## Description: 
Unable to include a reference to a video

### Actual Result:

-----

### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/5846

<div type='discussions-op-text'>

<sup>Originally posted by **johnmuchow** June  7, 2023</sup>
### Description

I am using the format {{< video local-video.mp4 >}} to include a video that is in the same folder as the .qmd file. I get the error message "The media could not be loaded, either because the server or network failed or because the format is not supported."

The file is an mp4. Is there any additional debug information available to help narrow down the issue?

</div>

**Comment by cderv** [comment]
@johnmuchow sorry the initial ping did not work correctly. Could you share some piece of code to help us reproduce this and look into it ?

**Comment by bryanhanson** [comment]
I'm having the same problem.  This zip contains a reproducible example.

[test2.zip](https://github.com/quarto-dev/quarto-cli/files/11929968/test2.zip)

**Comment by mcanouil** [comment]
@bryanhanson Would you mind sharing the example either as a repository or as flat files?

**Comment by bryanhanson** [comment]
No problem.

## _quarto.yml
```
project:
  title: "test"
  type: "website"
```

## test.qmd
````
---
title: "Trying to Embed Video"
---

We should see a video below:

{{< video slow-blink.mp4 >}}

Did it embed correctly? No.
````

Sample 6 second video attached.

```bash
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.2: OK
      Dart Sass version 1.55.0: OK
      Deno version 1.33.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      Path: /Users/bryanhanson/Documents/Professional/Research/R_Pkgs/quarto-cli/package/dist/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.2
      Path: /usr/local/opt/python@3.11/bin/python3.11
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library
      knitr: 1.43
      rmarkdown: 2.22

[✓] Checking Knitr engine render......OK

```

https://github.com/quarto-dev/quarto-cli/assets/140350/0b176915-e2d4-4af2-aed7-8e366f8d235e

**Comment by mcanouil** [comment]
I can reproduce this.
It happens only in projects and not in regular/simple HTML.

This can easily be "fixed" by adding the following:

```yaml
project:
  title: "test"
  type: "website"
  resources:
    - slow-blink.mp4
```

**Comment by bryanhanson** [comment]
Thank you!

**Comment by mcanouil** [comment]
To note, there might be something to do here in path resolution inside Quarto.

**Comment by bryanhanson** [comment]
Follow up: this problem originally occurred for me in a blog project.  Where do I put `resources:` within the folder of a single post?

**Comment by mcanouil** [comment]
Exactly as I showed and as documented in <https://quarto.org/docs/websites/website-tools.html#site-resources>.

**Comment by bryanhanson** [comment]
Got it, thanks, it works.  What threw me was it seems slightly counter-intuitive to put a resource for a single post in a file controlling the entire site, since all other images etc are within a particular post's folder.  No worries, much appreciated.

**Comment by mcanouil** [comment]
@dragonstyle will (I think) know more on `resources` and path resolution, which seems to be the issue here.

**Comment by cderv** [comment]
I believe this could be duplicates or at least related to 

* https://github.com/quarto-dev/quarto-cli/issues/5496
* https://github.com/quarto-dev/quarto-cli/issues/3892

And I think this is all related to this that needs to be fixed 
* https://github.com/quarto-dev/quarto-cli/issues/5268

**Comment by dragonstyle** [comment]
I agree with @cderv - we envisioned `resources` as a last resort tool for providing website/blog resources that aren't explicitly linked to from pages. We thought this would be pretty unusual and hence a global set of `globs` would be rarely used. I think once we fix the auto-discovery of `src` there will be no need for you to report this as a resource, as it should be...

**Comment by dragonstyle** [comment]
Sorry meant to auto-close this with the commit message, thx for pinpointing the issue (as always!) and the cleanup @cderv

**Comment by schwa021** [comment]
> I can reproduce this. It happens only in projects and not in regular/simple HTML.
> 
> This can easily be "fixed" by adding the following:
> 
> ```yaml
> project:
>   title: "test"
>   type: "website"
>   resources:
>     - slow-blink.mp4
> ```

FYI - this solution only works if you know the file in advance.

In my situation, the video file is determined in code, hence I can't add it as a resource.

Any solution in this case?

**Comment by cscheid** [comment]
> Any solution in this case?

It should be working in your case if you're using a standard HTML element to reference the video (because your code emits the element, and our resource discovery happens _after_ code execution).

**Comment by schwa021** [comment]
> > Any solution in this case?
> 
> It should be working in your case if you're using a standard HTML element to reference the video (because your code emits the element, and our resource discovery happens _after_ code execution).

@cscheid - thank you for replying, and I apologize in advance for my ignorance.

For starters - I barely understand half the words you wrote. I don't know what a "standard HTML element" is, I don't know what you mean by my code "emit[ting] the element", and I don't know what you mean by "resource discovery".

Here is what I do know - I had code that worked in Rmarkdown and in quarto it gives me the error: 

> "The media could not be loaded, either because the server or network failed or because the format is not supported."

My code looks like this (resides within a chunk with results: asis)

new.vid is a string containing the path to the video

```
cat("<video width=640 controls>
  <source src=", new.vid, " type=video/mp4>
  </video>")
```
If new.vid is in the working directory or a subdirectory (e.g. "VIDEOS/new.vid") it works.
If new.vid is on a network drive (e.g., "\\\\something\\Videos\\somethingelse\\2023\\xxxxxx.mp4") I get the error.

I thought it might be the nagging issue with the wrong-direction slashes - but even if I hardcode a network drive using "/" it does not work.

Again - I apologize in advance for (probably) wasting your time. It's difficult on github.com/issues when you are just a hoi polloi type user.

**Comment by cscheid** [comment]
> If new.vid is in the working directory or a subdirectory (e.g. "VIDEOS/new.vid") it works.
If new.vid is on a network drive (e.g., "\\something\Videos\somethingelse\2023\xxxxxx.mp4") I get the error.

Ok. I think this the cause of the problem, and it's a place where quarto and RMarkdown simply work differently. In quarto, a resource file needs to be in the "same project", which means "in a folder that's a subfolder of _quarto.yml (or a "sub-sub-folder", etc)

An HTML "element" is a "piece" of html. `<a href="https://www.google.com">link</a>` is an HTML element. So is `<b>this is bold</b>`, and so on.

Your code that is doing

```
cat("<video width=640 controls>
  <source src=", new.vid, " type=video/mp4>
  </video>")
```

is "emitting a `video` element". You'll need to put the video into the subdirectory of your quarto project, and I think then things should work.

If they don't, then it might be a bug of some sort in quarto, in which case you should open a new issue. We'll need a full reproducible example to be able to diagnose further, though.

**Comment by matiasandina** [comment]
I was having a lot of trouble with this issue. Things I tried:

* short code
* pure HTML
* changing the path of the video to subfolders

Nothing seemed to work until I decided to

* Try a youtube video <- worked fine with shortcode
* Try *another* video <- worked fine with all my previous attempts

It turned out the issue was that my video of interest had `MPEG-4 Video (Simple Profile)` as the codec and, as far as I understood, this codec is not compatible with web browsers.
I am documenting this here just in case somebody else finds this issue. My quarto version is 1.4.55
